llvm/lib/Target/RISCV/RISCVRegisterInfo.td

```diff
@@ -752,18 +752,24 @@ def VR : VReg<!listconcat(VM1VTs, VMaskVTs),
 
 def VRNoV0 : VReg<!listconcat(VM1VTs, VMaskVTs), (sub VR, V0), 1>;
 
+let AllocationPriority = 2 in
```

Comment:
Can we do this in side of `VReg` using the lmul?

Reply:
In the current change, yes.  I'd done it this way because I'd originally planned to have the NoV0 cases have different values.  I'll switch.

Reply:
Doing this revealed that I hadn't handled the segment tuple register classes.  I decided to treat those as having the same priority as their lmul component; that is, I ignored NF.  There might be a better heuristic here.